### PR TITLE
Gvrf controls sample

### PR DESCRIPTION
**Pull request by Maginador**
GVRf joystick controls sample created by Samsung SIDIA.

GearVRf-DCO-1.0-Signed-off-by: Ricardo Maginador
ricardo.cm@samsung.com

### CODE REVIEW

File: GVRf/Sample/gvrf_controls_sample/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+
+
+    <application
+        android:allowBackup="true"
+        android:icon="@drawable/gearvr_logo"
+        android:label="@string/app_name"
+        android:theme="@style/AppTheme" >
+           <!--  <meta-data android:name="com.samsung.android.vr.application.mode" android:value="vr_only"/> -->
```
Comment (thomasflynn) [inline GVRf/Sample/gvrf_controls_sample/AndroidManifest.xml]: please uncomment this.  thanks.

File: GVRf/Sample/gvrf_controls_sample/res/values/strings.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */-->
+
+<resources>
+
+    <string name="app_name">GVR Simple Sample</string>
```
Comment (thomasflynn) [inline GVRf/Sample/gvrf_controls_sample/res/values/strings.xml]: please call this something other than "simple sample".

**Comment by thomasflynn** [comment]
Nice.  Here's a few things I noticed.  I haven't looked into the code to determine if any of these are intentional:

1) All the buttons on the controller light up when pressed except R1, Start, and Select.
2) The little character moves up/down/left/right with the DPad, but only moves up and left with the analog sticks.

**Comment by Maginador** [comment]
Fixed the issues.

**Comment by thomasflynn** [comment]
excellent.  one last question:  is the controller supposed to be behind the viewer when they start (meant as a surprise?) or is it meant to be in front?

**Comment by Maginador** [comment]
That's the expected behavior. So the user will find new objects as he looks around.

**Comment by thomasflynn** [comment]
ok, cool.  looked at the code.  it seems this commit also includes the keyboard sample.  was that intended??

**Comment by Maginador** [comment]
No, both samples were in the same branch, Keyboard was sent accidentally with Control.

**Comment by thomasflynn** [comment]
so, you're going to remove it, right?

**Comment by Maginador** [comment]
Yes I'll remove it as soon as possible and give the feedback here.

File: GVRf/Sample/gvrf_controls_sample/assets/gvr_note4.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */-->
+
+
+
+<lens name="N4">
+    <scene fov-y="90.0" camera-separation-distance="0.062" fbo-width="1024" fbo-height="1024" msaa="1"/>
+</lens>
```
Comment (chenchao1407) [inline GVRf/Sample/gvrf_controls_sample/assets/gvr_note4.xml]: We just changed the format of xml configurations. Please follow other samples to see how it is set now.

Thanks in advance.

File: GVRf/Sample/gvrf_controls_sample/src/org/gearvrf/controls/ChainElement.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        float nextY = target.getTransform().getPositionY() -
+                currentY;
+
+        float nextZ = target.getTransform().getPositionZ() -
+                currentZ;
+
+        getTransform().setPosition(
+                currentX + nextX, currentY + nextY, currentZ + nextZ);
+
```
Comment (chenchao1407) [inline GVRf/Sample/gvrf_controls_sample/src/org/gearvrf/controls/ChainElement.java]: Not quite sure why you need to do the computation.

currentX + nextX = target.getTransform().getPositionX() - currentX + currentX = target.getTransform().getPositionX().

Same with currentY + nextY, currentZ + nextZ.

File: GVRf/Sample/gvrf_controls_sample/src/org/gearvrf/controls/util/VRSamplesTouchPadGesturesDetector.java [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    }
+    
+//    private void showLogsAtPref(){
+//        
+//        for(int i = 0; i < 60; i++){
+//            
+//            Log.d(DEBUG_TAG, " - " + sharedPref.getFloat("SWIPE_DISTANCE"+i, -00000));
+//        }
+//    }
+
```
Comment (chenchao1407) [inline GVRf/Sample/gvrf_controls_sample/src/org/gearvrf/controls/util/VRSamplesTouchPadGesturesDetector.java]: remove commented codes, thanks

**Comment by chenchao1407** [comment]
Tried with controls sample without a controller, looks fine to me. And still waiting for you to remove the keyboard sample.

Thanks

**Comment by Maginador** [comment]
Removed Keyboard commit from this branch, changed xml structure as asked, removed commented code and removed ChainElement.java, it is not used inside the project.

**Pull request merged**